Docs/source/How_To/coding-style.rst

```diff
@@ -0,0 +1,270 @@
+*************
+Coding Style
+*************
+
+.. note:: 
+
+    | CamelCase - FirstWordSecondWord
+    | camelCase - firstWordSecondWord
+
+Macros and Constants
+====================
+
+Should be written in snakecase with all caps. If a macro utilizes arithmetic, it should be covered in parentheses to prevent compiler
+from assuming priority. Ex
+
+.. code-block:: c
+
+    #define ALL_CAPS_WITH_UNDERSCORE
+    #define math (4 + x)
+    const uint8_t FOUR_KG = 4;
+    #ifdef TASKS_H //No leading underscores
+
+Enums
+=====
+
+Enum name should be in CamelCase and all element in enum should be all caps. They should be located in the header file. Ex:
+
+.. code-block:: c
+
+    enum ExampleEnum {
+        ELEMENT_ONE;
+        ELEMENT_TWO;
+        ELEMENT_THREE;
+        };
+
+Structs
+=======
+
+All struct names should have a descriptive name written in CamelCase and an appended ``_t``. 
+Members of struct should be written in camelCase. They should be located in the header file. Ex:
+
+.. code-block:: c
+
+    typedef struct {
+        uint8_t element;
+        uint16_t elementTwo;
+    } StructName_t;
```

Comment:
I would recommend encouraging naming of structs, not just their typedef.

```suggestion
    typedef struct StructName {
        uint8_t element;
        uint16_t elementTwo;
    } StructName_t;
```

This can result in better gcc error messages when referring to this type.